### ISSUE TITLE
Restart issue for HydroFractureSolver

### ISSUE THREAD

**Issue by frankfeifan** [GEOS-DEV/GEOS#3286]
**Describe the bug**
I recently found an issue in restart when running the `HydroFractureSolver`. The following screenshot shows the meshes around the fracture for `pennyShapedToughnessDominated_smoke`. The left shows the non-restart initial run, and the right shows the restart run at the same time. There is apparently an issue in node ordering in the restart run. Here I only show the `CellElementRegion`, and looks like `SurfaceElementRegion` is fine. 
![Screenshot 2024-08-19 at 10 01 20 AM](https://github.com/user-attachments/assets/5f761b49-b58e-416e-990d-011c8dd98eed)

This issue can also be seen in the following screenshots of logs at the first timestep of restart. The left shows the non-restart initial run, and the right shows the restart run at the same time. Apparently the apertures calculated are different between two cases. Also, the convergence is affected in the following solve steps. In some more complicated problems I'm running, we could eventually get non-convergence. 
![Screenshot 2024-08-19 at 10 21 58 AM](https://github.com/user-attachments/assets/91949c8f-5b88-4b89-b895-43364b75f979)

The above result is for the case with partition of 2 in the crack normal direction (x in this case), but if we change the partition to 1 in the crack normal direction, it seems the issue is gone. See the following screenshot where the left is the restart run for the case with `-x 2 -y 2 -z 2` , while the right is the restart run for the case with `-x 1 -y 2 -z 2`. However, I don't think it permanently solves the issue. 
![Screenshot 2024-08-19 at 10 02 42 AM](https://github.com/user-attachments/assets/ece2f966-3131-416b-a9f5-d0ffe98e9304)

**Platform:**
- Machine: LLNL/dane & ruby
- Compiler: gcc 12.1.1 & clang 14.0.6

**Additional context**
From the logs as shown above, we can see the logs of the nonlinear solver and the linear solver are identical between the non-restart and the restart case before fracture propagation, but the values of aperture become different. So the issue should happen between `nonlinearImplicitStep` and the field initialize and update for new fractures. After some testing, I think the issue probably happens in `SurfaceGenerator::solverStep`.

**Comment by frankfeifan** [comment]
I've tried `kgdToughnessDominated_smoke` in integrated test and turned on the restart check (see PR #3302). The test failed in restart for both serial and parallel runs. Attached are restart check logs for serial run and parallel run. Apparently, the issues come from `localToGlobalMap`. In serial run, it doesn't affect the solve much, but in parallel run, the global fields of variables can be messed up and eventually affect the solution. 

[0291.python_kgdToughnessDominated_smoke_01_5_restartcheck_.log](https://github.com/user-attachments/files/16723223/0291.python_kgdToughnessDominated_smoke_01_5_restartcheck_.log)
[0296.python_kgdToughnessDominated_smoke_06_5_restartcheck_.log](https://github.com/user-attachments/files/16723224/0296.python_kgdToughnessDominated_smoke_06_5_restartcheck_.log)

**Comment by frankfeifan** [comment]
I did more testing in serial run. The .hdf5 file output for the restart run looks good in `localToGlobalMap`. So the problem is probably when GEOS reads in the file and runs the restart.

The following screenshot shows a part of `localToGlobalMap` of `nodeManager` in the .hdf5 file at the restart step. The left one is output from the initial run and will be input to the restart run, while the right one is output from the restart run. We can see the local and global indices are correctly mapped in the left one, but in the right one, the mapping for new nodes (should be due to fracture propagation) are apparently incorrect. 

![Screenshot 2024-08-23 at 11 40 49 AM](https://github.com/user-attachments/assets/91ced2ca-c1f7-4005-9b80-33076aaaf691)

**Comment by paveltomin** [comment]
is this https://github.com/GEOS-DEV/GEOS/issues/1004 related or it is a separate issue?

**Comment by paveltomin** [comment]
@frankfeifan @CusiniM this issue can be closed since https://github.com/GEOS-DEV/GEOS/pull/3302 got merged, right?

**Comment by frankfeifan** [comment]
> @frankfeifan @CusiniM this issue can be closed since #3302 got merged, right?

Yes. I think it can be closed.
